tools/cassandra/handler.go

```diff
@@ -47,24 +48,48 @@ type SetupSchemaConfig struct {
 func VerifyCompatibleVersion(
 	cfg config.Persistence,
 ) error {
-
-	ds, ok := cfg.DataStores[cfg.DefaultStore]
-	if ok && ds.Cassandra != nil {
-		err := CheckCompatibleVersion(*ds.Cassandra, cassandra.Version)
-		if err != nil {
+	if ds, ok := cfg.DataStores[cfg.DefaultStore]; ok {
+		if err := verifyCompatibleVersion(ds, map[string]string{
+			cassandra_db.PluginName: cassandra.Version,
+		}); err != nil {
 			return err
 		}
 	}
-	ds, ok = cfg.DataStores[cfg.VisibilityStore]
-	if ok && ds.Cassandra != nil {
-		err := CheckCompatibleVersion(*ds.Cassandra, cassandra.VisibilityVersion)
-		if err != nil {
+
+	if ds, ok := cfg.DataStores[cfg.VisibilityStore]; ok {
+		if err := verifyCompatibleVersion(ds, map[string]string{
+			cassandra_db.PluginName: cassandra.VisibilityVersion,
```

Comment:
It's generally not a good practice to pass a map like this. If you have to, maybe define a new struct. But here I think we can keep passing the version only..

It's a bigger work for supporting different plugins. We could do that separately(feel free to do it if you are interested!)

Reply:
Yeah probably I can only pass in cassandra version here for now and return an error if plugin name is not equal to `cassandra`?

Reply:
Yes, that’s fine for now.

Reply:
Updated to only pass in cassandra version to the func.